src/tests/mmi.test.ts

```diff
@@ -0,0 +1,258 @@
+import { MoneyMarketIndex, USDC, WETH } from 'constants/tokens'
+import { QuoteToken } from 'quote/quoteToken'
+import { FlashMintQuoteProvider } from 'quote'
+import { WrappedQuoteProvider } from 'quote/wrapped'
+import { getFlashMintWrappedContract } from 'utils/contracts'
+import { wei } from 'utils/numbers'
+
+import {
+  approveErc20,
+  allowanceOf,
+  balanceOf,
+  LocalhostProvider,
+  SignerAccount2,
+  transferFromWhale,
+  wrapETH,
+} from './utils'
+
+const indexToken = MoneyMarketIndex
+const indexTokenAddress = indexToken.address!
+
+const provider = LocalhostProvider
+const signer = SignerAccount2
+
+const mmi: QuoteToken = {
+  address: indexToken.address!,
+  decimals: 18,
+  symbol: indexToken.symbol,
+}
+
+const usdc: QuoteToken = {
+  address: USDC.address!,
+  decimals: 6,
+  symbol: USDC.symbol,
+}
+
+const weth: QuoteToken = {
+  address: WETH.address!,
+  decimals: 18,
+  symbol: WETH.symbol,
+}
+
+describe('MMI (mainnet)', () => {
+  beforeEach((): void => {
+    jest.setTimeout(10000000)
+  })
+
+  test('can mint MMI from WETH (convenience)', async () => {
+    // Get quote
+    const quoteRequest = {
+      isMinting: true,
+      inputToken: weth,
+      outputToken: mmi,
+      indexTokenAmount: wei(1),
+      slippage: 0.5,
+    }
+    const quoteProvider = new FlashMintQuoteProvider(provider)
```

Comment:
The idea of this new `FlashMintQuoteProvider` is maximum convenience by returning a `tx` object which can be used to just sign and send it.

Reply:
Both way work for minting now.